### DISCUSSION TITLE
How to either create a hyper::Error or handle hyper::http::Error

### DISCUSSION THREAD

**Discussion by ristillu** [hyperium/hyper#2802] [Q&A] [answered]
Let's say I have code like this:
```
fn bad_request() -> Result<Response<Body>> {
    Ok(Response::builder()
            .status(StatusCode::BAD_REQUEST)
            .body(Body::empty())?)
}
```
This fails to compile with `From<hyper::http::Error> is not implemented for hyper::Error`. It feels to me that it would make sense if it was? Any suggestions what I can do about it? I mean, I can `unwrap()` but I'd rather `map_err` (in other analogous but way more extensive code) creating a `hyper::Error` myself. But I can't do that because the details are private. Or I could implement my own error handling, but I'd rather use what feels so close. I.e. my best case solution would be if `From` was implemented.

**Comment by seanmonstar** [comment, marked as answer]
A `hyper::Error` is not meant to be something that a user creates. You don't need to return one in your server handler. I imagine you probably want to because you also are handling request body errors? Here's what I would do: change your service error to a dynamic error type. Maybe `anyhow`, or even just `Box<dyn std::error::Error>`.

**Comment by ristillu** [comment]
Thanks Sean. My n00b-ness showing through. I've previously shied away from those two approaches, trying to roll my own as I learn better how Rust works. But I've seen enough of both suggestions before that it's probably high time to try them.
